master/internal/config/resource_manager_config.go

```diff
@@ -171,21 +175,64 @@ type KubernetesResourceManagerConfig struct {
 	DefaultComputeResourcePool string `json:"default_compute_resource_pool"`
 	NoDefaultResourcePools     bool   `json:"no_default_resource_pools"`
 
-	ExposeProxiesExternally *ExposeProxiesExternallyConfig `json:"expose_proxies_externally"`
+	InternalTaskGateway *InternalTaskGatewayConfig `json:"internal_task_gateway"`
 
 	Name     string            `json:"name"`
 	Metadata map[string]string `json:"metadata"`
 }
 
-// ExposeProxiesExternallyConfig is config for exposing Kubernetes proxies externally on a service.
+// InternalTaskGatewayConfig is config for exposing Determined tasks to outside of the cluster.
 // Useful for multirm when we can only be running in a single cluster.
-type ExposeProxiesExternallyConfig struct {
+type InternalTaskGatewayConfig struct {
 	GatewayName      string `json:"gateway_name"`
 	GatewayNamespace string `json:"gateway_namespace"`
-	// TODO(RM-267/gateways) should this be a service name or an ip?
-	// Also is address the right term? Do we refer to hostnames / addresses as address or ip?
-	// A load balancer would need a static ip for address to make sense. Is this ok?
-	GatewayAddress string `json:"gateway_address"`
+	GatewayIP        string `json:"gateway_ip"`
+	GWPortStart      int    `json:"gateway_port_range_start"`
+	GWPortEnd        int    `json:"gateway_port_range_end"`
+}
+
+var defaultInternalTaskGatewayConfig = InternalTaskGatewayConfig{
+	GWPortStart: 1025,
+	GWPortEnd:   65535,
+}
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (k *InternalTaskGatewayConfig) UnmarshalJSON(data []byte) error {
+	*k = defaultInternalTaskGatewayConfig
+	type DefaultParser *InternalTaskGatewayConfig
+	return json.Unmarshal(data, DefaultParser(k))
+}
+
+// Validate implements the check.Validatable interface.
+func (i *InternalTaskGatewayConfig) Validate() []error {
+	var errs []error
+
+	if err := check.IsValidK8sLabel(i.GatewayName); err != nil {
+		errs = append(errs, errors.Wrap(err, "invalid GatewayName"))
+	}
+
+	if err := check.IsValidK8sLabel(i.GatewayNamespace); err != nil {
+		errs = append(errs, errors.Wrap(err, "invalid GatewayNamespace"))
+	}
+
+	if err := check.IsValidIPV4(i.GatewayIP); err != nil {
```

Comment:
could this be a hostname too?

Reply:
I went with ip to signal this is a static address for a loadbalancer. we could have it be a hostname and mention that in the docs for now

Reply:
happy to add that in separately after if we decide we want it. let's chat [here](https://hpe-aiatscale.slack.com/archives/C06K1CD0DMW/p1716864615405799?thread_ts=1716338052.746629&cid=C06K1CD0DMW)